cylc/flow/task_outputs.py

```diff
@@ -609,15 +624,23 @@ def _is_compvar_complete(self, compvar: str) -> Optional[bool]:
         else:
             raise KeyError(compvar)
 
-    def iter_required_messages(self) -> Iterator[str]:
+    def iter_required_messages(
+        self,
+        exclude=None
+    ) -> Iterator[str]:
         """Yield task messages that are required for this task to be complete.
 
         Note, in some cases tasks might not have any required messages,
         e.g. "completion = succeeded or failed".
+
+        Args:
+            exclude: Exclude one possible required message, allowing
+            specification of all required outputs if succeeded or failed.
```

Comment:
Could this be explained better? I am not sure what this means (though judinging from the tests it is not as simple as just excluding the specified output from the return value, which was my first impression)

Reply:
I've had a go.